pkg/gator/validate/validate_test.go

```diff
@@ -0,0 +1,107 @@
+package validate
+
+import (
+	"testing"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/gator/fixtures"
+	y3 "gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+func TestValidate(t *testing.T) {
+	tcs := []struct {
+		name   string
+		inputs []string
+		want   int
```

Comment:
This feels too light of a check. Since we're just checking length, we don't know which ones returned validation errors that shouldn't have. While we do print the returned errors in the log, it doesn't say which validation errors we expected. Further - we don't know if the _wrong_ set returned errors, but it just happened to be the correct size and so the test passed.

Reply:
I've improved this to use `cmpopts.IgnoreFields` so it use test Results that have enough identifying characteristic for us to know the Results we're expecting are being produced.